### PR TITLE
Feature/adopt python grammar fixes

### PR DESCRIPTION
**Pull request by TwoOfTwelve**
#280 

This pr replaces the antlr grammar taken from [bkiers/python3-parser](https://github.com/bkiers/python3-parser) by the one provided in the official antlr repositories. Both implementations follow the same general structure and all changes mentioned in (#280) seem to be present in the new grammar to.

There are also new tests for the python language module, which work with the old implementation as well as the new one. Sonarcloud does report some code smells in the grammar, but I think, it is better to leave that code as it is in the official repository.

### CODE REVIEW

**Review by tsaglam — COMMENTED** [review]
Just a few things.

File: languages/python-3/src/main/java/de/jplag/python3/grammar/Python3LexerBase.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+                    this.emit(createDedent());
+                    indents.pop();
+                }
+            }
+        }
+    }
+
+    @Override
+    public void reset() {
+        tokens = new java.util.LinkedList<>();
```
[the remark this answers is not in the document]
Comment (tsaglam) [reply languages/python-3/src/main/java/de/jplag/python3/grammar/Python3LexerBase.java:141]: As these code files are not that long and not complicated a ANTLR grammar, I think we can properly clean them up and then add a javadoc comment that states where this code was token from. So feel free to optimize the code.

[the remark this answers is not in the document]
Comment (dfuchss) [reply languages/python-3/src/main/java/de/jplag/python3/grammar/Python3LexerBase.java:141]: We could also think about using git-subtree in order to use external code easily

**Comment by tsaglam** [comment]
@TwoOfTwelve only the sonar code smells left; then I can merge.

**Review by tsaglam — APPROVED** [review]

**Comment by tsaglam** [comment]
@TwoOfTwelve did this PR change the supported language version of Python? Or is it still 3.6?

**Pull request merged**